website/source/docs/guides/connect-gateways.md

```diff
@@ -0,0 +1,421 @@
+to maintain quorum.
+
+```json
+connect {
+  "enabled": true
+}
+```
+
+The `primary_datacenter` setting that was required in order to enable ACL
+replication between datacenters also specifies which datacenter will write
+intentions and act as the [root CA for Connect](LINK TO MULTI-DC CONNECT DOCS).
+Intentions, which allow or deny inter-service communication, are automatically
+replicated to the secondary datacenter.
+
+## Deploy Gateways
+
+Connect mesh gateways proxy requests from services in one datacenter to services
+in another, so you will need to deploy your gateways on nodes that can reach
+each other over the network. As we mentioned in the prerequisites,
+you will need to make sure that both Envoy and Consul are installed on the
+gateway nodes. You won’t want to run any services on these nodes other than
+Consul and Envoy because they necessarily will have access to the WAN.
+
+### Generate Tokens for the Gateways
+
+You’ll need to [generate a
+token](/consul/security-networking/production-acls#apply-individual-tokens-to-the-services)
+for each gateway that gives it read access to the entire catalog.
+
+Create a file named `mesh-gateway-policy.json` containing the following content.
+
+```json
+{
+  "service_prefix": {
+    "": {
+      "policy": "read"
+    }
+  }
+}  
+{
+  "service": {
+      "mesh-gateway": {
+        "policy": "write"
+      }
+    }
+}
+```
+
+Next, create and name a new ACL policy using the file you just made.
+
+```text
+$ consul acl policy create \
+  -name mesh-gateway \
+  -rules @mesh-gateway-policy.json
+```
+
+Generate a token for each gateway from the new policy.
+
+```text
+$ consul acl token create -description "mesh-gateway primary datacenter token" \
+  -policy-name mesh-gateway
+```
+
+```text
+$ consul acl token create \
+  -description "mesh-gateway secondary datacenter token" \
+  -policy-name mesh-gateway
+```
+
+You’ll apply those tokens when you deploy the gateways.
+
+### Deploy the Gateway for your primary datacenter
+
+Register and start the gateway in your primary datacenter with the following
+command.
+
+```text
+$ consul connect envoy -mesh-gateway -register \
+                     -service-name "gateway-primary"
+                     -address "<your private address>" \
+                     -wan-address "<your externally accessible address>"\
+                     -token=<token for the primary dc gateway>
+```
+
+### Deploy the Gateway for your Secondary Datacenter
+
+Register and start the gateway in your secondary datacenter with the following
+command.
+
+```text
+$ consul connect envoy -mesh-gateway -register \
+                     -service-name "gateway-secondary"
+                     -address "<your private address>" \
+                     -wan-address "<your externally accessible address>"\
+                     -token=<token for the secondary dc gateway>
+```
+
+### Configure Sidecar Proxies to use Gateways
```

Comment:
One thing to note is that for this to work you have to set `enable_central_service_config = true` in all the configuration of agents that will be managing sidecar proxies (although it would also be useful for all agents even ones only managing gateways).